_opensearch/rest-api/analyze-apis/perform-text-analysis.md

```diff
@@ -0,0 +1,670 @@
+
+<hr />
+
+You can create a custom transient normalizer with token and character filters.
+
+The following request uses the `uppercase` character filter to convert the given text to all upper case:
+
+````json
+GET /_analyze
+{
+  "filter" : ["uppercase"],
+  "text" : "That is the boy who followed me."
+}
+````
+
+The previous request returns the following fields:
+
+````json
+{
+  "tokens" : [
+    {
+      "token" : "THAT IS THE BOY WHO FOLLOWED ME.",
+      "start_offset" : 0,
+      "end_offset" : 32,
+      "type" : "word",
+      "position" : 0
+    }
+  ]
+}
+````
+
+#### Get token details
+
+You can obtain additional details for all tokens by setting the `explain` attribute to `true`.
+
+The following request provides detail token information for the `reverse` filter used with the `standard` tokenizer. 
+
+````json
+GET /_analyze
+{
+  "tokenizer" : "standard",
+  "filter" : ["reverse"],
+  "text" : "OpenSearch analyze test",
+  "explain" : true,
+  "attributes" : ["keyword"] 
+}
+````
+
+The previous request returns the following fields:
+
+````json
+{
+  "detail" : {
+    "custom_analyzer" : true,
+    "charfilters" : [ ],
+    "tokenizer" : {
+      "name" : "standard",
+      "tokens" : [
+        {
+          "token" : "OpenSearch",
+          "start_offset" : 0,
+          "end_offset" : 10,
+          "type" : "<ALPHANUM>",
+          "position" : 0
+        },
+        {
+          "token" : "analyze",
+          "start_offset" : 11,
+          "end_offset" : 18,
+          "type" : "<ALPHANUM>",
+          "position" : 1
+        },
+        {
+          "token" : "test",
+          "start_offset" : 19,
+          "end_offset" : 23,
+          "type" : "<ALPHANUM>",
+          "position" : 2
+        }
+      ]
+    },
+    "tokenfilters" : [
+      {
+        "name" : "reverse",
+        "tokens" : [
+          {
+            "token" : "hcraeSnepO",
+            "start_offset" : 0,
+            "end_offset" : 10,
+            "type" : "<ALPHANUM>",
+            "position" : 0
+          },
+          {
+            "token" : "ezylana",
+            "start_offset" : 11,
+            "end_offset" : 18,
+            "type" : "<ALPHANUM>",
+            "position" : 1
+          },
+          {
+            "token" : "tset",
+            "start_offset" : 19,
+            "end_offset" : 23,
+            "type" : "<ALPHANUM>",
+            "position" : 2
+          }
+        ]
+      }
+    ]
+  }
+}
+````
+
+#### Set a token limit
+
+You can set a limit to the number of tokens generated. Setting lower values reduces a node's memory usage. The default value is 10000.
+
+The following request limits the tokens to four:
+
+````json
+PUT /books2
+{
+  "settings" : {
+    "index.analyze.max_token_count" : 4
+  }
+}
+````
+The preceding request is an index API rather than an analyze API. See [DYNAMIC INDEX SETTINGS]({{site.url}}{{site.baseurl}}/opensearch/rest-api/index-apis/create-index/#dynamic-index-settings) for additional details.
```

Comment:
Is the preceding request literally an index API, or should there be some clarification here?